src/hotspot/share/runtime/globals.hpp

```diff
@@ -1997,6 +1997,10 @@ const int ObjectAlignmentInBytes = 8;
                                                                             \
   product(bool, StressSecondarySupers, false, DIAGNOSTIC,                   \
           "Use a terrible hash function in order to generate many collisions.") \
+                                                                            \
+  product(bool, UseThreadStartLock, true, DIAGNOSTIC,                       \
+          "Use an extra lock in JVM_ThreadStart to reduce "                 \
+          "time-to-safepoint while creating threads.")                      \
```

Comment:
I think the flag can be just `UseThreadStartLock`. The description should probably be more generic, e.g.:

```
Use an extra lock during Thread.start to alleviate contention on Threads lock.
```